src/Network/Receive.pm

```diff
@@ -3824,4 +3824,154 @@ sub hat_effect {
 	}
 }
 
+sub npc_talk_close {
+	my ($self, $args) = @_;
+	# 00b6: long ID
+	# "Close" icon appreared on the NPC message dialog
+	my $ID = $args->{ID};
+	my $name = getNPCName($ID);
+
+	message TF("%s: Done talking\n", $name), "npc";
+
+	# I noticed that the RO client doesn't send a 'talk cancel' packet
+	# when it receives a 'npc_talk_closed' packet from the server'.
+	# But on pRO Thor (with Kapra password) this is required in order to
+	# open the storage.
+	#
+	# UPDATE: not sending 'talk cancel' breaks autostorage on iRO.
+	# This needs more investigation.
+	#if (!$talk{canceled}) {
+	#	$messageSender->sendTalkCancel($ID);
+	#}
+	# Now Task::TalkNPC sends this by default
+
+	$ai_v{'npc_talk'}{'talk'} = 'close';
+	$ai_v{'npc_talk'}{'time'} = time;
+	undef %talk;
+
+	Plugins::callHook('npc_talk_done', {ID => $ID});
+}
+
+sub npc_talk_continue {
+	my ($self, $args) = @_;
+	my $ID = substr($args->{RAW_MSG}, 2, 4);
+	my $name = getNPCName($ID);
+
+	$ai_v{'npc_talk'}{'talk'} = 'next';
+	$ai_v{'npc_talk'}{'time'} = time;
+	
+	message TF("%s: Type 'talk cont' to continue talking\n", $name), "npc";
```

Comment:
Maybe put this message into TalkNPC task as well (and display it only when `autoTalkCont` is not set, and also when there are no more steps so user interaction is needed)?

Actually, maybe display all of those "type ... to continue talking" messages only when there are no more steps in the current sequence, so they don't appear at all when conversations are fully automated?